### PR TITLE
[Verifier] verify-bytecode-meter prints module/function breakdown

### PR DESCRIPTION
**Pull request by amnn**
## Description

Use a custom meter to track every function and module that is verified, and display all of them.

## Test Plan

```
deepbook$ cargo run --bin sui -p sui \
  -- client verify-bytecode-meter    \
  --module build/DeepBook/bytecode_modules/math.mv

╭──────────────────────────────────────╮
│ Package will pass metering check!    │
├──────────────────────────────────────┤
│ Limits                               │
├─────────────────────────┬────────────┤
│ packages                │ 16,000,000 │
│   modules               │ 16,000,000 │
│     functions           │ 16,000,000 │
├─────────────────────────┴────────────┤
│ Ticks Used                           │
├─────────────────────────┬────────────┤
│ <unknown>               │     41,915 │
│   math                  │     41,915 │
│     unsafe_mul          │      1,225 │
│     unsafe_mul_round    │      5,675 │
│     mul                 │      2,410 │
│     mul_round           │      2,905 │
│     unsafe_div          │      1,225 │
│     unsafe_div_round    │      6,085 │
│     div_round           │      2,905 │
│     count_leading_zeros │     19,485 │
├─────────────────────────┴────────────┤
│ Package will pass metering check!    │
╰──────────────────────────────────────╯
```

## Stack

- #16903
- #16941 
- #16945 

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [x] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### Release notes

`sui client verify-bytecode-meter` prints a breakdown of verifier cost by module and function in a package.

### CODE REVIEW

File: crates/sui/src/client_commands.rs [hunk 1]
```diff
@@ -2079,42 +2098,91 @@ impl Display for SuiClientCommandResult {
...
+
+                let message = if *success {
+                    "Package will pass metering check!"
+                } else {
+                    "Package will NOT pass metering check!"
+                };
+
+                // Add overall header and footer message;
+                table.with(TablePanel::header(message));
+                table.with(TablePanel::footer(message));
```
[the remark this answers is not in the document]
Comment (stefan-mysten) [reply crates/sui/src/client_commands.rs:2166]: I would say that this is even desired, having both footers and headers for long tables is useful!

File: crates/sui/src/client_commands.rs [hunk 2]
```diff
@@ -2079,42 +2098,91 @@ impl Display for SuiClientCommandResult {
...
+
+                let mut table = builder.build();
+
+                let message = if *success {
+                    "Package will pass metering check!"
+                } else {
+                    "Package will NOT pass metering check!"
+                };
+
+                // Add overall header and footer message;
```
[the remark this answers is not in the document]
Comment (stefan-mysten) [reply crates/sui/src/client_commands.rs:2164]: I remember fiddling with it too, but I don't have any tips on top of my mind. I think in the newer version of tabled things are a bit cleaner, but there's a lot of breaking changes and did not get around updating everything.

File: crates/sui/src/client_commands.rs [hunk 3]
```diff
@@ -1115,22 +1125,31 @@ impl SuiClientCommands {
...
                 verifier.meter_compiled_modules(&protocol_config, &modules, &mut meter)?;
 
+                let mut used_ticks = meter.accumulator(Scope::Package).clone();
+                used_ticks.name = pkg_name;
+
                 let meter_config = protocol_config.meter_config();
+
+                let exceeded = matches!(
+                    meter_config.max_per_pkg_meter_units,
+                    Some(ticks) if ticks < used_ticks.max_ticks(Scope::Package)
```
Comment (tnowacki) [inline crates/sui/src/client_commands.rs:1138]: nit just for quick reading maybe
```suggestion
                    Some(allowed_ticks) if allowed_ticks < used_ticks.max_ticks(Scope::Package)
```

**Review by dariorussi — APPROVED** [review]
it looks good to me and great break down of info

**Pull request merged**